examples/basic/01_async_data_transfer_client.py

```diff
@@ -1,138 +1,138 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""
+.. _ref_async:
+
+==================================
+Run file operations asynchronously
+==================================
+
+This script is intended to be run from the command line, where it
+authenticates with the specified HPS service and sets up an asynchronous
+client for data transfer operations. It then transfers files to and from remote backends
+using the data transfer service.
+
+Example usage:
+``python examples/async_data_transfer_client.py --url "https://example.com/hps" --username "user" --password "pass"``
+"""
+
+import asyncio
+import glob
+import logging
+import os
+
+import typer
+from typing_extensions import Annotated
+
+from ansys.hps.data_transfer.client import AsyncClient, AsyncDataTransferApi
+from ansys.hps.data_transfer.client.authenticate import authenticate
+from ansys.hps.data_transfer.client.models.msg import SrcDst, StoragePath
+
+log = logging.getLogger(__name__)
+logger = logging.getLogger()
+logging.basicConfig(format="%(asctime)s %(levelname)8s > %(message)s", level=logging.DEBUG)
+
+
+async def main(
+    debug: Annotated[bool, typer.Option(help="Enable debug logging")] = False,
+    url: Annotated[str, typer.Option(help="HPS URL to connect to")] = "https://localhost:8443/hps",
+    username: Annotated[str, typer.Option(help="Username to authenticate with")] = "repadmin",
+    password: Annotated[
+        str, typer.Option(prompt=True, hide_input=True, help="Password to authenticate with")
+    ] = "repadmin",
+):
+
+    dt_url = f"{url}/dt/api/v1"
+    auth_url = f"{url}/auth/realms/rep"
+    token = authenticate(username=username, password=password, verify=False, url=auth_url)
+    token = token.get("access_token", None)
+    assert token is not None
+
+    log.info("Connecting to the data transfer service client..")
+    client = AsyncClient(clean=True)
+
+    client.binary_config.update(
+        verbosity=3,
+        debug=debug,
+        insecure=True,
+        token=token,
+        data_transfer_url=dt_url,
+    )
+    await client.start()
+
+    ########################################
+    # Create a AsyncDataTransferApi instance
```

Comment:
Did you intend the use of this ReStructuredText formatting here? In the rendered doc, it looks odd within the code block.

Reply:
@PipKat, I agree. Some weeks ago I had made an attempt at fixing it and adding more of such formatting. Let me know what you think: https://github.com/ansys/pyhps-data-transfer/pull/200/files

Reply:
@saimanikant I looked at PR #200 and made some minor edits. I feel like we need to get these PRs merged and conflicts resolved. I can than do another pass at an overall doc review.

Reply:
Thanks @PipKat. Addressed your comments on that PR. I'll merge that once I get your sign off.

Reply:
@PipKat The PR for updating the examples doc is merged: https://github.com/ansys/pyhps-data-transfer/pull/200.
Thanks